Adding filter

Frenchris opened the pull request.

kigiri reviewed (changes requested):
so yeah, kind of the same remarks, ask for functions instead of values.

use monkeypatching to ensure they use filter in the tests

```js
// solutions would go from
let arr2 = arr1.filter((el) => el % 2 === 0)
// to
const multipleOf2 = arr => arr.filter((el) => el % 2 === 0)
```

Frenchris committed:
update of module for "filter" exercise to run

Frenchris committed:
subject for "filter" exercise updated

Frenchris committed:
updated solution for new "filter" exercise (only frenchris one is working)

Frenchris committed:
Updated filter subject and solution
1) made the difficulty more progressive from one function to the next one
2) transformed the data to get close to a real life example

kigiri committed:
call-me-maybe/filter: fix subject and rework tests

kigiri committed:
call-me-maybe/filter: pass kigiri
